### ISSUE TITLE
scrollable-region-focusable: false positive for disabled multiline select / listbox

### ISSUE THREAD

**Issue by MichaelKetting** [dequelabs/axe-core#3692]
### Product

axe-core

### Product Version

3.3.4 ... 4.4.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
scrollable-region-focusable should not report for multiline select elements / listbox (when they are diabled)

#### Actual
I have a disabled select element showing a scrollbar because it contains more items than than it has rows configured and I get scrollable-region-focusable in my aXe result.

#### How to Reproduce
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
<head>
<title>Example</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
</head>
<body>
  <select size="2" name="select" disabled="disabled" aria-label="Test">
    <option value="First">First</option>
    <option value="Second">Second</option>
    <option value="Third">Third</option>
  </select>
</body>
</html>

```

#### Additional context
Since the browser does not allow focusing the element, it is not within the web-developer's scope to solve this issue for disabled elements.

see also #3691

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base,
The `Scrollable-region-focussible` rule is not failing when `select` `size` makes scrollable for the list of items.
Test snippet:
```
<select size="2" name="select" disabled="disabled" aria-label="Test">
    <option value="First">First</option>
    <option value="Second">Second</option>
    <option value="Third">Third</option>
  </select>
```



![Image](https://user-images.githubusercontent.com/47409579/214528224-5b412be0-6393-4c43-a8f6-222918fde818.png)
